### PR TITLE
[SCB-300] acceptance test for @Compensable event timeout

### PR DESCRIPTION
**Pull request by lijasonvip**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Now we have acceptance test for `@SagaStart` event timeout in #171 , we need to test `@Compensable` event timeout too.

### CODE REVIEW

File: acceptance-tests/acceptance-pack/src/test/resources/hotel_timeout.btm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+IF TRUE
+DO RETURN 2
+ENDRULE
+
+RULE sleep when postHotelBooking until timeout happens
+CLASS org.apache.servicecomb.saga.demo.pack.hotel.HotelBookingService
+METHOD postHotelBooking
+AT ENTRY
+IF TRUE
+DO debug("delay 10s until the car booking timeout"),
```
Comment (WillemJiang) [inline acceptance-tests/acceptance-pack/src/test/resources/hotel_timeout.btm]: Debug log is not right

File: acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    Then Alpha records the following events
+      | serviceName  | type               |
+      | pack-booking | SagaStartedEvent   |
+      | pack-car     | TxStartedEvent     |
+      | pack-car     | TxEndedEvent       |
+      | pack-hotel   | TxStartedEvent     |
+      | pack-hotel   | TxAbortedEvent     |
+      | pack-car     | TxCompensatedEvent |
+      | pack-hotel   | TxCompensatedEvent |
+      | pack-hotel   | SagaEndedEvent     |
```
Comment (WillemJiang) [inline acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature]: I don't think pack-hotel should send the SagaEndedEvent here.

File: acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      | pack-hotel   | TxEndedEvent       |
+      | pack-booking | SagaEndedEvent     |
+
+    Then Car Service contains the following booking orders
+      | name  | amount | confirmed | cancelled |
+      | Jason | 1      | false     | true      |
+
+    And Hotel Service contains the following booking orders
+      | name  | amount | confirmed | cancelled |
+      | Jason | 1      | true      | false     |
```
Comment (WillemJiang) [inline acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature]: The Hotel Service should be cancelled.  There must be some errors in the Alpha Service.

File: acceptance-tests/acceptance-pack/src/test/resources/pack_car_timeout_scenario.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      | pack-booking | SagaStartedEvent   |
+      | pack-car     | TxStartedEvent     |
+      | pack-car     | TxAbortedEvent     |
+      | pack-car     | TxCompensatedEvent |
+      | pack-car     | SagaEndedEvent     |
+      | pack-car     | TxEndedEvent       |
+
+    Then Car Service contains the following booking orders
+      | name  | amount | confirmed | cancelled |
+      | Jason | 1      | true      | false     |
```
Comment (WillemJiang) [inline acceptance-tests/acceptance-pack/src/test/resources/pack_car_timeout_scenario.feature]: The Car Service should be cancelled.

**Comment by WillemJiang** [comment]
It looks like the Accept test result is not good. We need to fix the issue here.

**Comment by zhfeng** [comment]
what is the purpose of this test ? I think it would like to show the scenario that when the saga transaction is timeout, the alpha server call the compensate method is failed and will retry ?

The booking service looks like the following steps:
1) saga start
2) call the car booking
3) call the hotel booking
4) postBooking wait 10s until the transaction timeout (injected by the byteman rules)
5) return failure message with the information of timeout ( I think it is still an issue when we are sending the SAGA_ENDED in the https://github.com/apache/incubator-servicecomb-saga/blob/master/omega/omega-transaction/src/main/java/org/apache/servicecomb/saga/omega/transaction/SagaStartAnnotationProcessor.java#L45). I think we should check the return value and throws the Exception if the transaction is timeout. Also the alpha server handles the message in (https://github.com/apache/incubator-servicecomb-saga/blob/master/alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/TxConsistentService.java#L38), currently it only check if the transaction is aborted and I think it should also check if the transaction is timeout.

For this test, if we want to simulate the alpha server retries the compensate method when the transaction is timeout, I think it would like to use the byteman rule in the cancel method of HotelBooking (https://github.com/apache/incubator-servicecomb-saga/blob/master/saga-demo/booking/hotel/src/main/java/org/apache/servicecomb/saga/demo/pack/hotel/HotelBookingService.java#L40)
```
RULE create counter
CLASS org.apache.servicecomb.saga.demo.pack.hotel.HotelBooking
METHOD <init>
IF TRUE
DO createCountDown($0, 1)
ENDRULE

RULE throws the exception when calling the cancel method at the first time
CLASS org.apache.servicecomb.saga.demo.pack.hotel.HotelBookingService
MEHTOD cancel(HotelBooking)
AT ENTRY
IF !countDown($0)
DO throw exception
ENDRULE
```

File: acceptance-tests/acceptance-pack/src/test/resources/hotel_timeout.btm (hunk 1 above)
Comment (lijasonvip) [reply acceptance-tests/acceptance-pack/src/test/resources/hotel_timeout.btm]: My bad.

File: acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature (hunk 2 above)
Comment (lijasonvip) [reply acceptance-tests/acceptance-pack/src/test/resources/pack_hotel_timeout_scenario.feature]: I'll fix this.

**Pull request closed without merging**